### DISCUSSION TITLE
k8s-dns-node-cache Matrix with CoreDNS?

### DISCUSSION THREAD

**Discussion by scotttrevors** [coredns/coredns#4901] [Q&A] [answered]
Hi,
Are versions of CoreDNS and k8s-dns-node-cache tied to each other?  

IE do you have to be on a certain version of CoreDNS to use the lastest k8s-dns-node-cache?

CoreDNS 1.8.0 - Can i use 1.21.1 k8s-dns-node-cache?

**Comment by chrisohaver** [comment, marked as answer]
> IE do you have to be on a certain version of CoreDNS to use the lastest k8s-dns-node-cache?

no, they are independent.

More detail: CoreDNS (the cluster service discovery service) and k8s-dns-node-cache (the node local dns cache instances) just forward standard dns messages to each other, which are not tied to any version.

**Reply by scotttrevors** [reply]
Thanks. Is there a dependency between CoreDNS and the K8S API? Can we upgrade to the latest version of "CoreDNS" on an older version of Kubernetes?

**Reply by chrisohaver** [reply]
It should work on older versions of the API.
